lightly/transforms/random_crop_and_flip_with_grid.py

```diff
@@ -158,58 +167,36 @@ def forward(self, img: Image.Image) -> ImageTensorAndGrid:
+        return img, grid
 
     def location_to_NxN_grid(self, location: Location) -> torch.Tensor:
 
         """
-        Create a grid tensor with `N` rows and `N` columns, where each cell represents a region of
+        Create a grid tensor with grid_size rows and grid_size columns, where each cell represents a region of
         the original image. The grid is used to map the cropped and transformed image back to the
         original image space.
 
         Args:
-            location: An instance of the `Location` class, containing the location and size of the
+            location: An instance of the Location class, containing the location and size of the
                 transformed image in the original image space.
 
         Returns:
-            A grid tensor of shape (N, N, 2), where the last dimension represents the (x, y) coordinate
+            A grid tensor of shape (grid_size, grid_size, 2), where the last dimension represents the (x, y) coordinate
             of the center of each cell in the original image space.
         """
         
-        size_h_case = location.h / self.N
-        size_w_case = location.w / self.N
-        half_size_h_case = size_h_case / 2
-        half_size_w_case = size_w_case / 2
-        final_grid_x = torch.zeros(self.N, self.N)
-        final_grid_y = torch.zeros(self.N, self.N)
-
-        final_grid_x[0][0] = location.i + half_size_h_case
-        final_grid_y[0][0] = location.j + half_size_w_case
-        for k in range(1, self.N):
-            final_grid_x[k][0] = final_grid_x[k - 1][0] + size_h_case
-            final_grid_y[k][0] = final_grid_y[k - 1][0]
-        for l in range(1, self.N):
-            final_grid_x[0][l] = final_grid_x[0][l - 1]
-            final_grid_y[0][l] = final_grid_y[0][l - 1] + size_w_case
-        for k in range(1, self.N):
-            for l in range(1, self.N):
-                final_grid_x[k][l] = final_grid_x[k - 1][l] + size_h_case
-                final_grid_y[k][l] = final_grid_y[k][l - 1] + size_w_case
-
-        final_grid = torch.stack([final_grid_x, final_grid_y], dim=-1)
-        if location.flip:
-            # start_grid = final_grid.clone()
-            for k in range(0, self.N):
-                for l in range(0, self.N // 2):
-                    swap = final_grid[k, l].clone()
-                    final_grid[k, l] = final_grid[k, self.N - 1 - l]
-                    final_grid[k, self.N - 1 - l] = swap
-        return final_grid
+        cell_width = location.width / self.grid_size 
+        cell_height = location.height / self.grid_size
+        x = torch.linspace(location.left, location.left + location.width, self.grid_size) + (cell_width / 2)
+        y = torch.linspace(location.top, location.top + location.height, self.grid_size) + (cell_height / 2)
+        if location.horizontal_flip:
+            x = torch.flip(x, dims=[0])
+        if location.vertical_flip:
+            y = torch.flip(y, dims=[0])
```

Comment:
Oh nice! Didn't know about `torch.flip`!

Reply:
It is less efficient, but indexing like you did is not possible in torch, only in numpy